src/libappimage/desktop_integration/integrator/ResourcesExtractor.cpp

```diff
@@ -11,49 +11,114 @@ namespace appimage {
             ResourcesExtractor::ResourcesExtractor(const core::AppImage& appImage) : appImage(appImage) {}
 
             DesktopIntegrationResources ResourcesExtractor::extract() {
+                /* Build a cache of the entries contained in the AppImage payload. Include the entries path,
+                 * type and in case of a link link entry the link target. Solve links chains in order to ease
+                 * the lookup in the next step.
+                 */
+                PayloadEntriesCache entriesCache(appImage);
+
+                /*
+                 * Use the cache to identify the resources entries that must be extracted. In case of LINK entries
+                 * store the final link target which must point to a REGULAR entry to ensure a proper extraction
+                 * by means of `PayloadIterator::read`.
+                 */
                 DesktopIntegrationResources resources = {};
-                for (auto fileItr = appImage.files(); fileItr != fileItr.end(); ++fileItr) {
-                    if (fileItr.type() != core::PayloadEntryType::REGULAR &&
-                        fileItr.type() != core::PayloadEntryType::LINK)
-                        continue;
+                resolveResourcesFinalPaths(resources, entriesCache);
+
+                /*
+                 * Call `PayloadIterator::read` on each entry included in resources and store the value in the
+                 * `data` members or in the right side of the maps.
+                 */
+                readResourceFiles(resources);
+
+                if (entriesCache.getEntryType(".DirIcon") == core::PayloadEntryType::LINK) {
```

Comment:
Seems like there should be some sort of `class Entry` which exposes things like type and maybe contents...

Reply:
This is something to discuss at #65